controllers/auth_config_controller.go

```diff
@@ -66,6 +66,16 @@ func (r *AuthConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	} else if errors.IsNotFound(err) || !Watched(&authConfig.ObjectMeta, r.LabelSelector) {
 		// could not find the resouce: 404 Not found (resouce must have been deleted)
 		// or the resource misses required labels (i.e. not to be watched by this controller)
+		apiConfig := r.Cache.Get(req.String())
+		if nil != apiConfig {
+			for _, identityConifg := range apiConfig.IdentityConfigs {
+				if v, ok := identityConifg.(common.AuthConfigCleaner); ok {
+					if err := v.Clean(ctx); err != nil {
```

Comment:
Tricky one, as it should be abstracted at this level, but we know from the implementation that each identity config should have no more than one evaluator and therefore no more than one cleaner. Maybe breaking the loop after successful cleaning once to improve performance?